independent-projects/arc/runtime/src/main/java/io/quarkus/arc/RequestContext.java

```diff
@@ -104,6 +126,10 @@ public void activate(Collection<ContextInstanceHandle<?>> initialState) {
             }
         }
         currentContext.set(state);
+        // Fire an event with qualifier @Initialized(RequestScoped.class) if there are any observers for it
+        if (!initializedNotifier.get().isEmpty()) {
```

Comment:
`LazyValue.get()` is always a `volatile` read and we should try to make it as fast as possible. I'd probably create a util method like this:
```java
static void fireIfNotEmpty(LazyValue<Notifier<Object>> value) {
   Notifier notifier = value.get();
   if (!notifier.isEmpty()) {
      notifier.notify(toString());
   }
}
```
WDYT?

Reply:
That's fair point, I've altered the code.
Though I am not sure why do you want the util method to be `static`?

Reply:
Ah sorry, of course it doesn't have to be `static`...

Reply:
Thought so ;)
It's private now and hence doesn't need the secondary parameter.